movement/analysis/kinematics.py

```diff
@@ -165,3 +169,472 @@ def compute_time_derivative(data: xr.DataArray, order: int) -> xr.DataArray:
+    dim : str
+        The dimension to compute the distances for. Must be either
+        ``'individuals'`` or ``'keypoints'``.
+    metric : str, optional
+        The distance metric to use. Must be one of the options supported
+        by :func:`scipy.spatial.distance.cdist`, i.e.
+        ``'braycurtis'``, ``'canberra'``, ``'chebyshev'``, ``'cityblock'``,
+        ``'correlation'``, ``'cosine'``, ``'dice'``, ``'euclidean'``,
+        ``'hamming'``, ``'jaccard'``, ``'jensenshannon'``, ``'kulczynski1'``,
+        ``'mahalanobis'``, ``'matching'``, ``'minkowski'``,
+        ``'rogerstanimoto'``, ``'russellrao'``, ``'seuclidean'``,
+        ``'sokalmichener'``, ``'sokalsneath'``, ``'sqeuclidean'``, ``'yule'``.
+        Defaults to ``'euclidean'``.
+    **kwargs : dict
+        Additional keyword arguments to pass to
+        :func:`scipy.spatial.distance.cdist`.
+
+    Returns
+    -------
+    xarray.DataArray
+        An xarray DataArray containing the computed distances between
+        each pair of inputs.
+
+    Examples
+    --------
+    Compute the Euclidean distance (default) between ``ind1`` and
+    ``ind2`` (i.e. interindividual distance for all keypoints)
+    using the ``position`` data variable in the Dataset ``ds``:
+
+    >>> pos1 = ds.position.sel(individuals="ind1")
+    >>> pos2 = ds.position.sel(individuals="ind2")
+    >>> ind_dists = cdist(pos1, pos2, dim="individuals")
+
+    Compute the Euclidean distance (default) between ``key1`` and
+    ``key2`` (i.e. interkeypoint distance for all individuals)
+    using the ``position`` data variable in the Dataset ``ds``:
+
+    >>> pos1 = ds.position.sel(keypoints="key1")
+    >>> pos2 = ds.position.sel(keypoints="key2")
+    >>> key_dists = cdist(pos1, pos2, dim="keypoints")
+
+    Obtain the distance between ``key1`` of ``ind1`` and
+    ``key2`` of ``ind2`` from ``ind_dists``
+    (i.e. interindividual distance, different keypoints):
+
+    >>> dist_ind1key1_ind2key2 = ind_dists.sel(ind1="key1", ind2="key2")
+
+    Equivalently, the same distance can be obtained from ``key_dists``:
+
+    >>> dist_ind1key1_ind2key2 = key_dists.sel(key1="ind1", key2="ind2")
+
+    Obtain the distance between ``key1`` and ``key2`` of ``ind1``
+    (i.e. interkeypoint distance within the same individual):
+
+    >>> dist_ind1key1_ind1key2 = key_dists.sel(key1="ind1", key2="ind1")
+
+    Obtain the distance between ``key1`` of ``ind1`` and ``ind2``
+    (i.e. interindividual distance, same keypoint):
+
+    >>> dist_ind1key1_ind2key1 = ind_dists.sel(ind1="key1", ind2="key1")
+
+    See Also
+    --------
+    scipy.spatial.distance.cdist : The underlying function used.
+    compute_interindividual_distances : Compute distances between one or
+        more pairs of individuals within and across all keypoints.
+    compute_interkeypoint_distances : Compute distances between one or
+        more pairs of keypoints within and across all individuals.
```

Comment:
I would have the taglines in "Seel also" match the one-line descriptions of the references functions.

See my other comment regarding the "within and across" phrasing.